pycsob/utils.py

```diff
@@ -121,3 +121,31 @@ def get_card_provider(long_masked_number):
         if rx.match(long_masked_number[:6]):
             return provider_id, conf.CARD_PROVIDERS[provider_id]
     return None, None
+
+
+def to_camel_case(value):
+    """
+    If the value is not None, convert the value from snake_case to camelCase format. If the value is not in snake_case
+    format, return the original value.
+    """
+
+    if value is None:
+        return value
+
+    first_word, *other_words = value.split('_')
+    return "".join([first_word.lower(), *map(str.title, other_words)]) if other_words else first_word
+
+
+def convert_keys_to_camel_case(data):
```

Comment:
Chová se to docela zvláštně, podle toho jestli tam dáš list nebo dict tak to pak vrací list nebo dict. Tohle se bude aji těžko typovat. Snažil bych se to sjednotit na 1 datovou strukturu.

Reply:
Mně to funguje dobře. typ je dict[str, Any]. Má to convertovat POUZE klíče z dictu, takže když tam dáš list, tak ti to má vrátit hodnoty tak jak jsou, proto i ve jméně key. 

Pokud tam dám list, který obsahuje dict, tak to vrátí list s dictem 🤔

Reply:
když tam dáš list stringů, tak to nemá nic udělat, to je expected 👍

Reply:
```
def convert_keys_to_camel_case(data: list[Any]) -> list[Any]:
    if not data:
        return data
    
    result = []
    for el in data:
        if isinstance(el, dict):
            result.append(
                {to_camel_case(key): value for key, value in el.items()}
            )
        elif isinstance(el, list):
            result.append(convert_keys_to_camel_case(el))
        else:
            log.warning("Invalid value")
            result.append(el)
        
    return result
```
řešení pouze s listem, více řádků, ale žádný copy paste a jasný interface, není tam mapping mezi typem argumentu a návratovou hodnotou. Nicméně ber to jako nitpick :-)

Reply:
Tohle řešení nefunguje úplně spolehlivě a hlavně by se s tím pak špatně a neintuitivně pracovalo pro případ, který potřebujeme. Jestli nevadí, nechal bych takto 🙏

Reply:
the function should accept list and dict and return the same type it is called with, typing can be solved through generics

```python
T = TypeVar("T", list, dict)
def convert_keys_to_camel_case(data: T) -> T:
    ...
```

Reply:
Good idea with generics :-) Agreed